lib/buttons-and-links/KButtonGroup.vue

```diff
@@ -0,0 +1,38 @@
+<template>
+
+  <span class="button-group">
+    <slot></slot>
+  </span>
+
+</template>
+
+
+<script>
+
+  /**
+   * Used to create a consistently spaced horizontal set of buttons and/or links.
+   *
+   * KButtonGroup works with KButton, KIconButton, KRouterLink and KExternalLink and any <a> or <button> tags.
+   * in it's default slot.
+   */
+
+  export default {
+    name: 'KButtonGroup',
+  };
+
+</script>
+
+
+<style lang="scss" scoped>
+
+  .button-group {
+    overflow: hidden;
+  }
+
+  /* KButton can produce <a> or <button> tags */
+  .button-group > a,
+  .button-group > button {
+    margin: 8px;
+  }
```

Comment:
I think this should only be adding internal margins, not external.

Something like

```css
  .button-group > a:not(:first-child) ,
  .button-group > button:not(:first-child)  {
    margin-left: 8px;
  }
```

(untested)